snp_tpp/src/roi_selection_mesh_modifier.cpp

```diff
@@ -5,38 +5,45 @@
 #include <tf2_eigen/tf2_eigen.h>
 #endif
 #include <tf2/time.h>
+#include <geometry_msgs/msg/point32.hpp>
 
 namespace snp_tpp
 {
 ROISelectionMeshModifier::ROISelectionMeshModifier(rclcpp::Node::SharedPtr node,
                                                    noether::ExtrudedPolygonSubMeshExtractor extractor,
-                                                   std::vector<geometry_msgs::msg::PointStamped> boundary)
+                                                   std::vector<geometry_msgs::msg::PolygonStamped> boundary)
   : boundary_(std::move(boundary)), extractor_(std::move(extractor)), buffer_(node->get_clock()), listener_(buffer_)
 {
 }
 
 std::vector<pcl::PolygonMesh> ROISelectionMeshModifier::modify(const pcl::PolygonMesh& mesh) const
 {
-  if (boundary_.size() < 3)
-  {
-    return { mesh };
-  }
-  else
+  std::vector<pcl::PolygonMesh> modifiedMeshes;
+
+  for(const geometry_msgs::msg::PolygonStamped& inner_vector_ : boundary_)
   {
-    Eigen::MatrixX3d boundary(boundary_.size(), 3);
-    for (Eigen::Index i = 0; i < boundary.rows(); ++i)
+    if (inner_vector_.polygon.points.size() < 5) // 5 because one single point/dot counts as 2, so a triangle (3 points in space) will count as size 6
+    {
+      std::cout << "***** Polygon points size is less than 3, size is: " << inner_vector_.polygon.points.size()/2 << " *****" << std::endl;
+      continue;
+    }
```

Comment:
Can you explain this in more detail? I don't understand why a polygon would need to have more than 3 points minimum

Reply:
Fixed double point generation by updating rviz_polygon_selection_tool repo, where I added an rviz property tool to define the gap between the last generated point and the new one the user wants to generate.
Commit id for scan_n_plan: [1d40097](https://github.com/ros-industrial-consortium/scan_n_plan_workshop/pull/114/commits/1d4009742dacbebb3b21d117dbabd3f3cac43f77)
Commit id for rviz_polygon_selection_tool: [79744be](https://github.com/BryanMqz/rviz_polygon_selection_tool/commit/79744be1fee7e84d035e32777875cd3b654e55be)